pkg/cmd/dev/test.go

```diff
@@ -105,12 +110,20 @@ pkg/kv/kvserver:kvserver_test) instead.`,
 	testCmd.Flags().String(testArgsFlag, "", "additional arguments to pass to the go test binary")
 	testCmd.Flags().String(vModuleFlag, "", "comma-separated list of pattern=N settings for file-filtered logging")
 	testCmd.Flags().Bool(showDiffFlag, false, "generate a diff for expectation mismatches when possible")
+	testCmd.Flags().String(volumeFlag, "bzlhome", "the Docker volume to use as the container home directory (only used for roachprod-stress cross builds)")
+	testCmd.Flags().String(clusterFlag, "", "the name of the roachprod cluster to use with roachprod-stress")
 	return testCmd
 }
 
 func (d *dev) test(cmd *cobra.Command, commandLine []string) error {
 	pkgs, additionalBazelArgs := splitArgsAtDash(cmd, commandLine)
 	ctx := cmd.Context()
+
+	roachprodCluster := mustGetFlagString(cmd, clusterFlag)
+	stress := mustGetFlagBool(cmd, stressFlag)
+	if stress && roachprodCluster != "" {
```

Comment:
This could be made better. We ought to integrate with as many of the parameters below --filter, --stress-args, --count. When run under stress for ex. we disable test sharding, that's logic we should inherit. We also seem to exclude disallowed_imports_test, which presumably applies to roachprod-stress as well. There's logic around the timeout parameter when combined with stress, which also applies perhaps, and streaming test output. There's existing code around parsing the package being addressed that we don't need to duplicate in roachprod_stress.go.

Point being, try using as much of what's applicable below so it composes in the way folks expect.

Reply:
I agree that merging the commands makes it seem like the entire API of `dev test` should be supported. In fact the set of supported command-line flags is much smaller. If the two commands are going to live under the same top-level API then at the very least we should check that inapplicable arguments (like `--filter` or `--count`) are not set before kicking off the `roachprod-stress` process. But more likely it suggests that `roachprod-stress` and `test` should not live under the same top-level API.

Reply:
--filter is applicable, right? I was thinking of the following "unification", and forking out to the roachprodStress code paths once --stress + --cluster were specified, with inputs that include the other args.

```
dev test <pkg> [--stress] [--cluster <cluster>] [--stress-args <stress-args>] [--filter <regex>] [--race] [--volume <volume>]
```

This is all UX-related only, so if it's not worth the effort, I'm fine dropping it.

Reply:
I agree that this provides better UX but I also agree that having all of the remaining non-applicable flags (used with `test` but not `roachprod-stress`) might confuse users (which is the opposite of what we are looking for in this code change) and if we will return an error when any of those flags is passed then maybe stressing remotely doesn't belong here because that's a lot of flags. Ideally, I'd love to see stressing locally and stressing remotely behave exactly the same. Instead of trying to let stressing remotely (`dev roachprod-stress`) fit in (`dev test`), we can:

1. Remove `dev roachprod-stress`
2. Move stressing locally to a separate command `dev stress`
3. Move stressing remotely to the same command `dev stress`